### PR TITLE
NP-38975 Add older license versions to dropdown

### PR DESCRIPTION
**Pull request by joachimjorgensen**
# Description

Link to Jira issue: [NP-38975](https://unit.atlassian.net/browse/NP-38975)

Add older license versions to the dropdown menu, hidden by "...". 

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

[NP-38975]: https://unit.atlassian.net/browse/NP-38975?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/registration/files_and_license_tab/FilesTableRow.tsx [hunk 1]
```diff
@@ -230,24 +237,48 @@ export const FilesTableRow = ({ file, removeFile, baseFieldName, showFileVersion
...
+                    dense
+                    sx={{ gap: '1rem', display: inactiveLicensesOpen ? 'flex' : 'none' }}>
+                    <ListItemIcon>
+                      <img style={{ width: '5rem' }} src={license.logo} alt={license.name} />
+                    </ListItemIcon>
+                    <ListItemText>
+                      <Typography>{license.name}</Typography>
+                    </ListItemText>
+                  </MenuItem>
+                ))}
```
Comment (joachimjorgensen) [inline src/pages/registration/files_and_license_tab/FilesTableRow.tsx:281]: Prøvde på å trekke ut dette til en felles komponent, men ble bare mye kluss da det er mange parametre som må sendes inn, og options må være likt strukturert for at verdier skal kunne velges (når det trekkes ut til en komponent så aksepteres ikke opsjonene lengre sånn uten videre).

**Review by denektenina — DISMISSED** [review]
Maybe it would be better with text instead of the three dots? It's not so intuitive to understand what the dots mean and since they are staying in the middle of the dropdown it could maybe have a role as a section headline of some sort. But not important so looks good from my side.

File: src/pages/registration/files_and_license_tab/FilesTableRow.tsx [hunk 2]
```diff
@@ -230,24 +237,48 @@ export const FilesTableRow = ({ file, removeFile, baseFieldName, showFileVersion
...
+                <MenuItem
+                  data-testid={dataTestId.registrationWizard.files.licenseItemShowOlderVersion}
+                  title={t('registration.files_and_license.show_all_older_versions')}
+                  sx={{ display: 'flex', justifyContent: 'center' }}
+                  onClickCapture={(e) => {
+                    e.stopPropagation();
```
Comment (elkebab) [inline src/pages/registration/files_and_license_tab/FilesTableRow.tsx:261]: er denne for å unngå at den lukkes ved trykk?

**Review by elkebab — DISMISSED** [review]
Har ikke testet, men koden ser bra ut!

Ellers enig med Nina, at det kunne vært en ide å vise en beskrivende tekst i stedet for "..."

File: src/pages/registration/files_and_license_tab/FilesTableRow.tsx [hunk 3]
```diff
@@ -230,24 +237,48 @@ export const FilesTableRow = ({ file, removeFile, baseFieldName, showFileVersion
...
+                      <img style={{ width: '5rem' }} src={license.logo} alt={license.name} />
+                    </ListItemIcon>
+                    <ListItemText>
+                      <Typography>{license.name}</Typography>
+                    </ListItemText>
+                  </MenuItem>
+                ))}
+                <MenuItem
+                  data-testid={dataTestId.registrationWizard.files.licenseItemShowOlderVersion}
+                  title={t('registration.files_and_license.show_all_older_versions')}
```
Comment (elkebab) [inline src/pages/registration/files_and_license_tab/FilesTableRow.tsx:258]: Denne blir forresten litt missvisende når bruker _har_ ekspandert feltet og faktisk viser gamle versjoner

File: src/pages/registration/files_and_license_tab/FilesTableRow.tsx (hunk 2 above)
Comment (joachimjorgensen) [reply src/pages/registration/files_and_license_tab/FilesTableRow.tsx:261]: Jepp. Vanlig onClick fanger ikke boblingen, men onClickCapture gjør det.

**Review by elkebab — APPROVED** [review]

**Pull request merged**